nixos/modules/virtualisation/oci-containers.nix

```
@@ -223,6 +247,13 @@ let
 
     preStart = ''
       ${cfg.backend} rm -f ${name} || true
+      ${optionalString (container.login.username != null && container.login.passwordFile != null) ''
+        cat ${container.login.passwordFile} | \
+          ${cfg.backend} login \
+            ${if container.login.registry != null then container.login.registry else ""} \
```

What registry `podman login` defaults to will depend on what is set in `registries.conf`.

We could require `container.login.registry` for the podman backend but it's probably simpler to just require that all three `container.login` fields are set.
```suggestion
      ${optionalString (container.login.username != null && container.login.passwordFile != null && container.login.registry != null) ''
        cat ${container.login.passwordFile} | \
          ${cfg.backend} login \
            ${container.login.registry} \
```